magma/backend/mlir/hardware_module.py

```diff
@@ -670,6 +673,50 @@ def visit_array_slice(self, module: ModuleWrapper) -> bool:
     def visit_when(self, module: ModuleWrapper) -> bool:
         return WhenCompiler(self, module).compile()
 
+    def visit_multi_port_memory(self, module: ModuleWrapper) -> bool:
+        inst = module.module
+        defn = type(inst)
+
```

Comment:
let's get rid of these whitespaces. if we want we can add one comment at the top that describes the implementation of this function